shipyard_blueprints/ftp/shipyard_ftp/cli/upload.py

```diff
@@ -45,77 +43,40 @@ def get_args():
 def main():
     try:
         args = get_args()
-        host = args.host
-        port = args.port
-        username = args.username
-        password = args.password
+
         source_file_name = args.source_file_name
         source_folder_name = args.source_folder_name
         destination_filename = args.destination_file_name or source_file_name
         destination_folder_name = shipyard.clean_folder_name(
             args.destination_folder_name
         )
-        source_file_name_match_type = args.source_file_name_match_type
-        errors = []
-        ftp_client = FtpClient(host=host, port=port, user=username, pwd=password)
-        if source_file_name_match_type == "exact_match":
-            source_full_path = shipyard.combine_folder_and_file_name(
-                folder_name=f"{os.getcwd()}/{source_folder_name}",
-                file_name=source_file_name,
-            )
-            destination_full_path = shipyard.determine_destination_full_path(
-                destination_folder_name=destination_folder_name,
-                destination_file_name=destination_filename,
-                source_full_path=source_full_path,
-            )
+
+        ftp_client = FtpClient(
+            host=args.host, port=args.port, user=args.username, pwd=args.password
+        )
+        files = shipyard.file_match(
```

Comment:
I'm assuming this `file_match` raises the `FileNotFound`?

Reply:
correct- here is what is happening under the hood for a sanity check

```
    matches = []
    source_directory = clean_folder_name(source_directory)

    if match_type == "exact_match":
        logger.debug(f"Searching for files in {files}...")
        if search_term not in files:
            raise FileNotFoundError(f"File {search_term} not found in {files}")

        source_full_path = combine_folder_and_file_name(source_directory, search_term)

        filename = determine_destination_full_path(
            destination_folder_name=destination_directory,
            destination_file_name=destination_filename,
            source_full_path=source_full_path,
        )

        matches.append(
            {"source_path": source_full_path, "destination_filename": filename}
        )

    elif match_type == "regex_match":
        matching_files = find_all_file_matches(files, search_term)

        if not matching_files:
            raise FileNotFoundError(f"No files found matching the regex {search_term}")

        logger.info(f"{len(matching_files)} files found.")
        for index, file_name in enumerate(matching_files, start=1):
            source_full_path = combine_folder_and_file_name(source_directory, file_name)
            filename = determine_destination_full_path(
                destination_folder_name=destination_directory,
                destination_file_name=destination_filename,
                source_full_path=file_name,
                file_number=index if len(matching_files) > 1 else None,
            )
            matches.append(
                {"source_path": source_full_path, "destination_filename": filename}
            )
    else:
        raise ValueError(
            f"Match type {match_type} is not supported. Supported types are 'exact_match' and "
            f"'regex_match'."
        )
    return matches
    ```